File: Part 02 - Regression/Section 6 - Polynomial Regression/polynomial_regression.R
# Polynomial Regression

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fitting linear regression
lin_reg = lm(formula = Salary ~ .,
             data = dataset)

# Fitting polynomial regression
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~.,
             data = dataset)

# Visualising results
# install.packages('ggplot2')
library(ggplot2)

ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)),
            colour = 'blue') +
  ggtitle('Truth or bluff (linear regression)') +
  xlab('Level') +
  ylab('Salary')

ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)),
            colour = 'blue') +
  ggtitle('Truth or bluff (polynomial regression)') +
  xlab('Level') +
  ylab('Salary')

# Predict results
y_pred = predict(lin_reg, newdata = data.frame(Level = 6.5))
y_pred = predict(poly_reg, newdata = data.frame(Level = 6.5,
                                                Level2 = 6.5^2,
                                                Level3 = 6.5^3,
                                                Level4 = 6.5^4))

File: Part 02 - Regression/Section 6 - Polynomial Regression/polynomial_regression.py
# Polynomial Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Fitting linear regression
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

# Fitting polynomial regression
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
lin_reg2 = LinearRegression()
lin_reg2.fit(X_poly, y)

# Visualising results
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))

plt.title('Truth or bluff')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.scatter(X, y, color = 'red')
# Linear regression
plt.plot(X, lin_reg.predict(X))
# Polynomial regression
plt.plot(X_grid, lin_reg2.predict(poly_reg.fit_transform(X_grid)))

# Predicting a result
y_pred_lin = lin_reg.predict(6.5)
y_pred_poly = lin_reg2.predict(poly_reg.fit_transform(6.5))